src/manual_folds_with_custom_loss.py

```diff
@@ -0,0 +1,635 @@
+        workers=workers,
+    )
+    all_brute_results = pd.concat(
+        [all_brute_results, results_meta_info_dict["search_results_df"]]
+    )
+
+    all_brute_results.sort_values(by="loss", inplace=True)
+    all_brute_results.reset_index(drop=True, inplace=True)
+    # valid_results_df = all_brute_results.loc[all_brute_results["loss"] != np.inf, :].copy()
+    top_result_df = all_brute_results.loc[0:0, :]
+
+    return top_result_df, all_brute_results, True
+
+
+# %% start of code
+
+# load in data
+file_path = utils.find_full_path(
+    "2021-05-02_equation_paper_aspirational_data_reduced", ".csv"
+)
+all_data = pd.read_csv(
+    file_path,
+    usecols=[
+        "geomean_basal_rate",
+        "geomean_isf",
+        "geomean_weighted_cir",
+        "bmi",
+        "geomean_total_daily_carbs_in_chunk_outcomes",
+        "geomean_total_daily_insulin_dose_in_chunk_outcomes",
+    ],
+)
+
+all_data.rename(
+    columns={
+        "geomean_basal_rate": "BR",
+        "geomean_isf": "ISF",
+        "geomean_weighted_cir": "CIR",
+        "bmi": "BMI",
+        "geomean_total_daily_carbs_in_chunk_outcomes": "CHO",
+        "geomean_total_daily_insulin_dose_in_chunk_outcomes": "TDD",
+    },
+    inplace=True,
+)
+
+# remove any rows where this is a value <= 0
+clean_data = all_data[np.sum(all_data <= 0, axis=1) == 0].copy()
+clean_data.reset_index(drop=True, inplace=True)
+
+clean_data["X_intercept"] = 1
+clean_data["BASAL"] = clean_data["BR"] * 24
+clean_data["log_BASAL"] = np.log(clean_data["BASAL"] + LOG_CONSTANT)
+clean_data["log_BR"] = np.log(clean_data["BR"] + LOG_CONSTANT)
+clean_data["log_ISF"] = np.log(clean_data["ISF"] + LOG_CONSTANT)
+clean_data["log_CIR"] = np.log(clean_data["CIR"] + LOG_CONSTANT)
+clean_data["log_BMI"] = np.log(clean_data["BMI"] + LOG_CONSTANT)
+clean_data["log_CHO"] = np.log(clean_data["CHO"] + LOG_CONSTANT)
+clean_data["log_TDD"] = np.log(clean_data["TDD"] + LOG_CONSTANT)
+clean_data["1/TDD"] = 1/clean_data["TDD"]
+
+y_cols = ["BASAL", "log_BASAL", "ISF", "log_ISF", "CIR", "log_CIR"]
+
+x_cols = [
+    "X_intercept",
+    "BMI",
+    "log_BMI",
+    "CHO",
+    "log_CHO",
+    "TDD",
+    "log_TDD",
+    "1/TDD"
+]
+
+X = clean_data[x_cols]
+y = clean_data[y_cols]
+
+# break the data into 70% train and 30% test set
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=1,
+)
+
+# loop through each of the 3 independent variables
+for y_name in Y_VARIABLE_LIST:  # [["BASAL", "log_BASAL"]]:  #, ["ISF", "log_ISF"], ["CIR", "log_CIR"]]:
+    print(f"solving for the {y_name} equations")
+
+    # load in the right bounding parameters or fitting checks
+    if "BASAL" in y_name:
+        bounding_check_dictionary = make_condition_dicts("basal_fitting_checks")
+    elif "ISF" in y_name:
+        bounding_check_dictionary = make_condition_dicts("isf_fitting_checks")
+    elif "CIR" in y_name:
+        bounding_check_dictionary = make_condition_dicts("cir_fitting_checks")
```

Comment:
I think the `isf_fitting_checks` and `cir_fitting_checks` may not have been added to this PR! Let me know if I missed them.